File: solutions/kotlin/leap/2/src/main/kotlin/Leap.kt
import java.time.DateTimeException
import java.time.LocalDate

data class Year(val year: Int) {

//    val isLeap: Boolean = try {
//        val dt = LocalDate.of(year, 2, 29)
//        true
//    } catch (e: DateTimeException) {
//        false
//    }
    val isLeap: Boolean = (year % 400 == 0) || ((year % 100 != 0) && (year % 4 == 0));
}

File: solutions/scala/leap/2/src/main/scala/Leap.scala
import java.time.{LocalDate, Period}

object Leap {
  def leapYear(year: Int): Boolean = {
    //    (year % 400) == 0 || ((year % 100) != 0 && (year % 4) == 0)
    val fromDate = LocalDate.parse(s"$year-01-01")
    val toDate = LocalDate.parse(s"$year-12-31")
    println(toDate.toEpochDay - fromDate.toEpochDay)
    toDate.toEpochDay - fromDate.toEpochDay == 365
  }
}
